ui/app/components/app/connected-sites-list/connected-sites-list.container.js

```diff
@@ -4,16 +4,33 @@ import { compose } from 'recompose'
 import ConnectedSitesList from './connected-sites-list.component'
 import {
   showModal,
+  legacyExposeAccounts,
 } from '../../../store/actions'
 import {
   getRenderablePermissionsDomains,
   getPermissionsDomains,
+  getAddressConnectedToCurrentTab,
+  getSelectedAddress,
 } from '../../../selectors/selectors'
+import { getOriginFromUrl } from '../../../helpers/utils/util'
 
 const mapStateToProps = state => {
+  const addressConnectedToCurrentTab = getAddressConnectedToCurrentTab(state)
+  const { title, url } = state.appState.currentActiveTab || {}
+  let tabToConnect
+
+  if (!addressConnectedToCurrentTab && url && !url.match(/^chrome-extension:\/\/.+\/home.html#connected$/)) {
```

Comment:
This `chrome-extension` protocol is only used on Chrome. Firefox uses `moz-extension`, and other browsers probably use others.

Maybe we could use an allow list here instead? Something like this:
```suggestion
  if (!addressConnectedToCurrentTab && url && ['http', 'https'].includes((new URL(url)).protocol)) {
```

The last time we did something like this, here's the list of dapp protocols we accepted: `['http:', 'https:', 'dweb:', 'ipfs:', 'ipns:', 'ssb:']`. Though in practice it's possible only `http` and `https` are used with MetaMask now.

Reply:
right...

Reply:
maybe there is a simpler solution that is future proof against possible additional protocols

Reply:
The "Connect page" idea would make an allow list no longer necessary. We could validate the field, add warnings, etc.

That protocol list was used when we first rolled out privacy mode by default and had a home notification prompting for the manual connect. But when we moved it to the connections settings page, the protocol list was removed and we left it up to the user to know what they're doing.

Reply:
We could also look for our extension specifically (using `extension.runtime.id`). But then it still allows connecting to things like files, pictures, etc. which is also undesirable.

Reply:
@Gudahtt What do you think about e58a2821a?

Reply:
Sure, that works at least for ignoring _our_ UI. For the other cases where the button isn't desired, I guess it's fine to leave it up to the user to know what they're doing 🤷‍♂. This is probably at least as good of a solution as I suggested.